GradeAnalytics/src/UtilityFunctions.java

```diff
@@ -1,9 +1,79 @@
+import java.lang.Object;
+import java.awt.*;
+import javax.swing.*;
 
 /**
  * Contains a variety of utility functions.
  * @author Michael Cai
  *
  */
+
+public class JXCanvas extends Canvas {
+	int[] nums;
+	String[] cat;
+	String t;
+	int max = 0;
+	int width = 0;
+	int height = 0;
+
+public JXCanvas(int[] x, String[] y, String a) {
```

Comment:
We'll be using the Canvas in the Main function, you should be taking in a canvas as a part of the parameters.